### PR TITLE
[WebNN EP] Support NPU deviceType

### PR DESCRIPTION
**Pull request by Honry**

### CODE REVIEW

**Comment by Honry** [comment]
@fdwr, @guschmue, PTAL, thanks!

File: js/common/lib/inference-session.ts [hunk 1]
```diff
@@ -243,7 +243,7 @@ export declare namespace InferenceSession {
   }
   export interface WebNNExecutionProviderOption extends ExecutionProviderOption {
     readonly name: 'webnn';
-    deviceType?: 'cpu'|'gpu';
+    deviceType?: 'cpu'|'gpu'|'npu';
```
Comment (fdwr) [inline js/common/lib/inference-session.ts:246]: We may need another parameter in the future like a fallback device type, or to let this be a *list* of devices in priority order (as a hint/preference anyway), but this lets us at least *test* it within ORT.

**Review by fdwr — APPROVED** [review]

**Pull request merged**